src/validator/CreditCardValidator.java

```diff
@@ -0,0 +1,62 @@
+package validator;
+
+import java.util.ArrayList;
+
+public class CreditCardValidator implements CreditCardChecking {
+
+	@Override
+	public boolean checkValidity(CreditCardVendor ccv, String number) {
+		ArrayList<Integer> chars = ccv.getNumberOfChars();
+		ArrayList<String> masks = ccv.getIIN();
+		
+		boolean result1 = checkLengthCorrectness(number, chars);
```

Comment:
If checkLengthCorrectness will be false here (result1==false) we do not need to verify result2 and result3. Because in line 16 we combine the results with AND operator. So if we know beforehand that some value will be false, we can return the result beforehand without calculating additional methods.